internal/venice-client-common/src/test/java/com/linkedin/venice/utils/concurrent/CloseableThreadLocalTest.java

```diff
@@ -32,33 +31,29 @@ public void testGet() {
     }
   }
 
-  @Test
-  public void testClose() {
-    CloseableThreadLocal<CloseableClass> numbers = new CloseableThreadLocal<>(() -> new CloseableClass(5));
-    Set<CloseableClass> closeableObjects = new HashSet<>();
+  @Test(timeOut = 30)
+  public void testClose() throws InterruptedException {
     int threadPoolSize = 10;
     int numRunnables = 1000;
+
+    CloseableThreadLocal<CloseableClass> numbers = new CloseableThreadLocal<>(() -> new CloseableClass(5));
     ExecutorService executorService = Executors.newFixedThreadPool(threadPoolSize);
-    try {
-      for (int i = 0; i < numRunnables; i++) {
-        executorService.submit(() -> {
-          closeableObjects.add(numbers.get());
-        });
-      }
-    } finally {
-      executorService.shutdown();
-    }
+    Set<CloseableClass> closeableObjects = ConcurrentHashMap.newKeySet();
+    Set<Thread> threadIds = ConcurrentHashMap.newKeySet();
```

Comment:
In JDK8, it is possible that this is very inefficient. Let's use `ConcurrentSkipListSet` instead

Reply:
I don't think efficiency matters much in this particular case. Is `ConcurrentSkipListSet` faster than `ConcurrentHashMap`?

Reply:
This is the bug in `ConcurrentHashMap` - https://bugs.openjdk.org/browse/JDK-8161372. We created a VeniceConcurrentHashMap to help with this but it doesn't work for `newKeySet`

It might be okay to use `newKeySet` in tests